compiler/damlc/tests/daml-test-files/RecordConstructorCheck.daml

```diff
@@ -0,0 +1,8 @@
+-- Copyright (c) 2020, Digital Asset (Switzerland) GmbH and/or its affiliates.
+-- All rights reserved.
+
+-- @WARN Record type X has constructor Y with different name. This may cause problems with cross-SDK upgrades. Possible solution: Change the constructor name to X
+
+module RecordConstructorCheck where
+
+data X = Y {}
```

Comment:
Can we please have a test for `newtype A = B{b :: Bool}` or something similar as well?

Reply:
Yep! I added it in a separate PR since this one already got merged.